embassy-stm32/src/eth/generic_smi.rs

```diff
@@ -51,6 +51,8 @@ pub struct GenericSMI {
 
 impl GenericSMI {
     /// Construct the PHY. It assumes the address `phy_addr` in the SMI communication
+    ///
+    /// Set `phy_addr` to `0xFF` for automatic detection (only with `time` feature enabled)
```

Comment:
it's not very "rusty" that there's a special value that has different behavior. I'd suggest either an enum `enum PhyAddr{ Auto, Manual(u8) }` or two different constructors `new` and `new_auto`

Reply:
I would keep this as special value in internal code, add address bound assert `(addr < 32)` in `new` method and add `new_auto`. This value must be replaced to actual addr. in `phy_reset`, or code will panic.